pytorch_lightning/plugins/training_type/ddp_spawn.py

```diff
@@ -217,14 +220,18 @@ def new_process(self, process_idx: int, trainer: "pl.Trainer", mp_queue: SimpleQ
         # ensure that spawned processes go through teardown before joining
         trainer._call_teardown_hook()
 
-    def post_dispatch(self):
+    def post_dispatch(self, trainer: "pl.Trainer"):
         # restore main state with best weights
         best_path = self.mp_queue.get()
         last_path = self.mp_queue.get()
         self._results = self.mp_queue.get()
         # get the `callback_metrics` and set it to the trainer
         # only in case the user does not override it.
-        self.lightning_module.get_from_queue(self.mp_queue)
+        # TODO(@daniellepintz): Remove the else in v1.7
+        if is_overridden("get_from_queue", trainer.training_type_plugin, DDPSpawnPlugin):
+            self.get_from_queue(trainer, self.mp_queue)
+        else:
+            self.lightning_module.get_from_queue(self.mp_queue)
 
```

Comment:
i think we should flip this, such that we only call lightning module.get_from_queue if it's overridden. otherwise we default ot calling the plugin's get_from-queue

Reply:
good call